src/mame/drivers/ladybug.cpp

```diff
@@ -135,42 +143,82 @@ void dorodon_state::decrypted_opcodes_map(address_map &map)
 }
 
 
-void sraider_state::sraider_cpu1_map(address_map &map)
+void mrsdyna_state::mrsdyna_cpu1_map(address_map &map)
 {
-	map(0x0000, 0x5fff).rom();
-	map(0x6000, 0x6fff).ram();
-	map(0x7000, 0x73ff).w("video", FUNC(ladybug_video_device::spr_w));
-	map(0x8005, 0x8005).r(FUNC(sraider_state::sraider_8005_r));  // protection check?
-	map(0x8006, 0x8006).writeonly().share("sound_low");
-	map(0x8007, 0x8007).writeonly().share("sound_high");
-	map(0x9000, 0x9000).portr("IN0");
-	map(0x9001, 0x9001).portr("IN1");
-	map(0x9002, 0x9002).portr("DSW0");
-	map(0x9003, 0x9003).portr("DSW1");
-	map(0xd000, 0xd7ff).w("video", FUNC(ladybug_video_device::bg_w));
-	map(0xe000, 0xe000).nopw();  //unknown 0x10 when in attract, 0x20 when coined/playing
+	// LS138 @ J4
+	map(0x0000, 0x1fff).rom();	// 2764 at R4
+	map(0x2000, 0x3fff).rom();	// 2764 at N4
+	map(0x4000, 0x5fff).rom();	// 2764 at M4
+	// LS138 @ J4 and LS139 @ H4
+	map(0x6000, 0x67ff).ram();  // 6116 @ K3, also clk on PAL K2 (16R6, U001)
+	map(0x6800, 0x6fff).ram();  // 6116 @ M3
```

Comment:
Don’t split up handlers for contiguous ROM/RAM – it’s inefficient.

Reply:
in cases like this it seems better documentation this way, which is most important?

Reply:
You can still put which chip covers which range in a comment.